pennylane/ops/qubit/matrix_ops.py

```diff
@@ -250,11 +264,23 @@ def _parse_control_values(control_wires, control_values):
         else:
             raise ValueError("Alternative control values must be passed as a binary string.")
 
-        return control_int
+        padding_left = control_int * len(U)
+        padding_right = 2 ** len(total_wires) - len(U) - padding_left
+
+        interface = qml.math.get_interface(U)
+        left_pad = qml.math.cast_like(qml.math.eye(padding_left, like=interface), 1j)
+        right_pad = qml.math.cast_like(qml.math.eye(padding_right, like=interface), 1j)
+        return qml.math.block_diag([left_pad, U, right_pad])
+
+    @property
+    def control_wires(self):
+        return self.hyperparameters["control_wires"]
```

Comment:
We can probably remove this going forward, I think it is used in the drawer.

Reply:
and circuit compilation.

Reply:
Wait, why should we remove this?

> and circuit compilation.

Yep, in fact @rmoyard has added `target_wires` as well in #1712